core/container.go

```diff
@@ -833,7 +903,86 @@ func (container *Container) withMounted(target string, srcDef *pb.Definition, sr
 	return container.containerFromPayload(payload)
 }
 
-func (container *Container) updateRootFS(ctx context.Context, subdir string, fn func(dir *Directory) (*Directory, error)) (*Container, error) {
+func (container *Container) chown(
+	ctx context.Context,
+	gw bkgw.Client,
+	srcDef *pb.Definition,
+	srcPath string,
+	owner string,
+	opts ...llb.ConstraintsOpt,
+) (*pb.Definition, string, error) {
+	ownership, err := container.ownership(ctx, gw, owner)
+	if err != nil {
+		return nil, "", err
+	}
+
+	if ownership == nil {
+		return srcDef, srcPath, nil
+	}
+
+	var srcSt llb.State
+	if srcDef == nil {
+		// e.g. empty cache mount
+		srcSt = llb.Scratch().File(
+			llb.Mkdir("/chown", 0o700, ownership.Opt()),
```

Comment:
Sorry for the late review, head was buried in debugging other stuff, but I *think* if you specify a mount point in an exec at a path that doesn't exist yet, buildkit defaults to 0755. So using 0700 here might technically be a deviation.

I really doubt this will ever matter, so I don't care if we leave it as is, just bringing it up in case you think it matters.

Reply:
Woops! Fix: https://github.com/dagger/dagger/pull/5006